pyclam/manifold.py

```diff
@@ -608,6 +605,48 @@ def __repr__(self) -> str:
     def depth(self) -> int:
         return len(self.graphs) - 1
 
+    def distance(self, *, x1: Union[int, List[int], None], x2: Union[int, List[int], None], p1: Data = None, p2: Data = None) -> np.ndarray:
```

Comment:
I see now what you intended here, to override if point not in self.data, but this function should not do that. Call it `argdistance` if you want.

Reply:
perhaps implement checks on type, list of ints is indices, numpy array is a dedicated point. Something other than this param string

Reply:
Just had a thought.. check dimensions. 1D arrays are indices, 2D are points. State that in the docstring, and make this positional, that would be very satisfying